cdap-api/src/main/java/co/cask/cdap/api/security/securestore/SecureStoreMetadata.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.security.securestore;
+
+import java.util.Date;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Represents the metadata for the data stored in the Secure Store.
+ */
+public final class SecureStoreMetadata {
+
+  private static final String DESCRIPTION_FIELD = "description";
+  private static final String DESCRIPTION_DEFAULT = "";
+
+  private final String name;
+  private final String description;
+  private final long createdEpochMs;
+  private final Map<String, String> properties;
+
+  private SecureStoreMetadata(String name, String description, Date created, Map<String, String> properties) {
```

Comment:
Since this is private constructor, we don't need to use `Date` type, simply have the `of` method pass in `long` (`System.currentTimeMillis`).